src/lib/telemetry/TelemetryReporter/report.ts

```diff
@@ -0,0 +1,56 @@
+const initTime = process.hrtime()
+
+import { TelemetryMetaMetrics } from '../../metrics/MetricNames'
+import { SessionManager } from '../../session/SessionManager'
+import { TelemetryLocalStore } from '../TelemetryStore'
+import { PendingTelemetryDataManager } from './PendingTelemetryDataManager'
+import { TelemetryReporter } from './TelemetryReporter'
+
+const reportTelemetry = async () => {
+  const reportTime = process.hrtime()
+  const telemetryObjFilePath = process.argv[3]
+  const reporter = TelemetryReporter.getTelemetryReporter()
+  const { account, workspace, tokenObj } = SessionManager.getSessionManager()
+
+  if (!account || !workspace || !tokenObj.isValid()) {
+    await reporter.moveTelemetryFileToPendingData(telemetryObjFilePath)
+  } else {
+    await reporter.reportTelemetryFile(telemetryObjFilePath)
+    await reporter.sendPendingData()
+  }
+
+  PendingTelemetryDataManager.getSingleton().registerPendingMetaMetric(
+    TelemetryMetaMetrics.REPORT_TIME,
+    process.hrtime(reportTime)
+  )
+}
+
+const prepareNewPendingFiles = async () => {
+  const pendingDataManager = PendingTelemetryDataManager.getSingleton()
+  try {
+    await pendingDataManager.acquireLock()
+    await pendingDataManager.createPendingFiles()
+    await pendingDataManager.ensureTelemetryDirMaxSize()
+    await pendingDataManager.releaseLock()
+  } catch (err) {
+    console.error(err)
```

Comment:
Will console.log work inside report script?

Reply:
Unfortunately no, but I think at this point if we get an error while creating the pending files we don't have much to do, what do you think?

Reply:
Same as above :)